src/library/scala/sys/process/ProcessImpl.scala

```diff
@@ -93,7 +98,7 @@ private[process] trait ProcessImpl {
     def start()     = { futureThread ;() }
 
     protected lazy val (processThread, (futureThread, futureValue), destroyer) = {
-      val code = new SyncVar[Option[Int]]()
+      val code = new LinkedBlockingQueue[Option[Int]](1)
```

Comment:
with max capacity 1

Reply:
@som-snytt I think by this way we could ensure the same behavior like `SyncVar` if we want to deprecate it.

Reply:
Deprecation doesn't mean you have to remove all usages (today, or ever). I already spent time on sys.process recently, I don't want to open it up again, it's very unsatisfying (because it needs rewriting anyway).

Reply:
Yes, on another PR they said no sys module for 2.13 because of feature freeze. I don't think my other PR clashes? I've already pushed it out of my head. My only advice is to focus on what must happen for 2.13. I think deprecation is OK, and maybe changing data structures will be OK, but I caution that people always think it will be OK.